### PR TITLE
Fix snap builds

### LINKED ISSUE #366: Snap setup (can't access) doesn't appear in port 3000
Fresh Ubuntu 16.04

Snap installed
`snap install convos`

Can't access convos in browser. Port 3000 and localhost don't return anything.

### PR DESCRIPTION
**Pull request by battlemidget**
- Use core18
- Update the building section to be conformant with latest recommendations

Fixes #366

Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

### CODE REVIEW

File: snap/local/README.md [hunk 1]
```diff
@@ -7,9 +7,9 @@
 
 # Building and testing the snap
 
-Make sure that Convos is checked out in your `$HOME` directory:
-
-    $ ./snap/build.sh
+    - Install snapcraft `sudo snap install snapcraft --classic`
+    - Clone convos repo
+    - Run `snapcraft` to build the snap (this will install multipass for vm based building)
```
Comment (jhthorsen) [inline snap/local/README.md]: Nitpicking, but this looks a bit inconsistent. Can you remove the spaces in front of "-" in the list?

I guess the "Make sure that Convos is checked out in your `$HOME` directory" statement is completely outdated?

File: snap/local/README.md [hunk 2]
```diff
@@ -7,9 +7,9 @@
```
Comment (jhthorsen) [inline snap/local/README.md]: Why is the README and build.sh moved into `snap/local/`? Should they be deleted instead..?

**Comment by jhthorsen** [comment]
> if you hook up github with snapcraft.io, it'll use launchpad to automatically build snaps.

You said something like the above on IRC - Unfortunately, I don't understand what you're saying. Can you point me to a howto or something to set this up?

File: snap/local/README.md (hunk 1 above)
Comment (battlemidget) [reply snap/local/README.md]: Yea you don't need this to be specifically checked out in `$HOME`

File: snap/local/README.md (hunk 2 above)
Comment (battlemidget) [reply snap/local/README.md]: Here is error output from snapcraft:

```
The 'snap' directory is meant specifically for snapcraft, but it contains the following non-snapcraft-related paths, which is unsupported and will cause unexpected behavior:
- build.sh

If you must store these files within the 'snap' directory, move them to 'snap/local', which is ignored by snapcraft.
```

**Comment by battlemidget** [comment]
> You said something like the above on IRC - Unfortunately, I don't understand what you're saying. Can you point me to a howto or something to set this up?

There are a couple of ways to do this, one is through travis-ci:

https://tutorials.ubuntu.com/tutorial/continuous-snap-delivery-from-travis-ci#0

The other way is to hook it up through snapcraft.io directly:

https://snapcraft.io/build

This will auth against your github account and do the build automatically

File: snap/local/README.md (hunk 2 above)
Comment (jhthorsen) [reply snap/local/README.md]: Then please delete the files. I'll merge this PR after you've done that.

**Comment by jhthorsen** [comment]
I will add the snippet below to the README after we've merged this:

```
[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/convos)
```

**Comment by jhthorsen** [comment]
Thank you so much for the two links! I've set it up through https://snapcraft.io/build now, but it's failing since we haven't merged this PR :)

**Comment by jhthorsen** [comment]
Awesome! Thank you 👍

**Pull request merged**